common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/azureactivedirectory/AzureActiveDirectoryAuthorizationResult.java

```diff
@@ -22,7 +22,117 @@
 // THE SOFTWARE.
 package com.microsoft.identity.common.internal.providers.microsoft.azureactivedirectory;
 
+import android.content.Intent;
+import android.os.Bundle;
+
+import com.microsoft.identity.common.adal.internal.AuthenticationConstants;
+import com.microsoft.identity.common.adal.internal.util.StringExtensions;
+import com.microsoft.identity.common.exception.ErrorStrings;
+import com.microsoft.identity.common.internal.logging.Logger;
+import com.microsoft.identity.common.internal.providers.oauth2.AuthorizationErrorResponse;
+import com.microsoft.identity.common.internal.providers.oauth2.AuthorizationResponse;
 import com.microsoft.identity.common.internal.providers.oauth2.AuthorizationResult;
+import com.microsoft.identity.common.internal.util.StringUtil;
+
+import java.io.Serializable;
+import java.util.HashMap;
+
+import static com.microsoft.identity.common.adal.internal.AuthenticationConstants.Browser.RESPONSE_ERROR_CODE;
+import static com.microsoft.identity.common.adal.internal.AuthenticationConstants.Browser.RESPONSE_ERROR_MESSAGE;
+import static com.microsoft.identity.common.adal.internal.AuthenticationConstants.Browser.RESPONSE_FINAL_URL;
 
 public class AzureActiveDirectoryAuthorizationResult extends AuthorizationResult {
+
+    private static final String TAG = AzureActiveDirectoryAuthorizationResult.class.getSimpleName();
+
+    public AzureActiveDirectoryAuthorizationResult(int resultCode, final Intent data) {
+        if (data == null) {
+            AuthorizationErrorResponse errorResponse = new AuthorizationErrorResponse
+                    (AuthorizationErrorResponse.AUTHORIZATION_FAILED, AuthorizationErrorResponse.NULL_INTENT);
+            setAuthorizationErrorResponse(AuthorizationStatus.FAIL, errorResponse);
+            return;
+        }
+        final Bundle extras = data.getExtras();
+        final int requestId = extras.getInt(AuthenticationConstants.Browser.REQUEST_ID);
+        switch (resultCode) {
+
+            case AuthenticationConstants.UIResponse.BROWSER_CODE_CANCEL:
+                Logger.verbose(TAG, null, "User cancel the request in webview: " + requestId);
```

Comment:
Is this passing null as the `correlationId`? There's a two-arg constructor which uses the `correlationId` associated to the thread (if there is one....)